EnemyControl/EnemyControl.lua

```diff
@@ -145,14 +146,31 @@ function EnemyControl.ReadPreset() --Read current preset and create table of ene
     end
 end
 
-function EnemyControl.UpdatePools() -- Inject every non-empty biome of the current preset into the relevant biomes in EnemySets.lua
+function EnemyControl.UpdatePools() -- Inject every non-empty biome of the current preset into the relevant biomes in EnemySets.lua and update EncounterData.lua
+    EnemySets = ModUtil.Table.Copy( EnemyControl.VanillaSets )
     DebugPrint({Text = "Enemy preset: "..EnemyControl.config.EnemySetting})
-    for biome, pool in pairs(EnemyControl.EligibleEnemies) do
-        EnemyControl.Target = RCLib.EncodeEnemySet(biome)
-        EnemyControl.Pool = pool
-        ModUtil.Table.Replace(EnemySets[EnemyControl.Target], EnemyControl.Pool)
+    local target = nil
+    for biome, pool in pairs( EnemyControl.EligibleEnemies ) do
+        target = RCLib.EncodeEnemySet( biome )
+        EnemySets[target] = ModUtil.Table.Copy( pool )
         DebugPrint({Text = "Updated enemy pool for "..biome})
     end
+    for encounter, data in pairs( EncounterData ) do -- If EnemySets is updated without updating EncounterData to match, behaviour is unreliable. Inheritance first
+        if data.InheritFrom then
```

Comment:
can we run `ProcessDataInheritence` here? SGG framework level code should be used as much as possible for consistency.

Reply:
Agreed that this is better, changed in new commit